engine/gameobject/src/dmsdk/gameobject/gameobject.h

```diff
@@ -356,7 +357,8 @@ namespace dmGameObject
         PropertyVar m_Variant;
         float* m_ValuePtr;
         bool m_ReadOnly;
-        uint8_t m_ValueType : 1;
+        uint16_t m_ValueType : 1;
+        uint16_t m_ArrayLength : 15;
```

Comment:
The array length was added since we already have empty space to use, and to avoid doing another call into the property stack to query the array length. We could merge the readonly into the uint16, or shorten the uint16 to an uint8 with 1 and 7 bits for the type and length but I don't know what's best here 🤔

Reply:
I moved the readonly into a single bit of the 16b